implementations/rust/ockam/ockam_command/src/project_member/delete.rs

```diff
@@ -28,78 +30,147 @@ pub struct DeleteCommand {
+        else if self.all {
+            if !opts
+                .state
+                .is_identity_enrolled(&Some(identity.name()))
+                .await?
+            {
+                return Err(miette!(
+                        "You need to use an enrolled identity to delete all the members from a Project."
+                    ).into());
             }
-            (None, true) => {
-                if !opts
-                    .state
-                    .is_identity_enrolled(&Some(identity.name()))
-                    .await?
+            let self_identifier = identity.identifier();
+            let member_identifiers = authority_node_client.list_member_ids(ctx).await?;
+            if !member_identifiers.is_empty() {
+                opts.terminal.write_line(fmt_info!(
+                    "Found {} members in the Project {}",
+                    member_identifiers.len(),
+                    project_name
+                ))?;
+            }
+
+            let members_to_delete = member_identifiers
+                .into_iter()
+                .filter(|id| id != &self_identifier)
+                .collect::<Vec<_>>();
+
+            let pb = opts.terminal.progress_bar();
+            for identifier in members_to_delete.into_iter() {
+                if let Some(pb) = &pb {
+                    pb.set_message(format!("Trying to delete member {identifier}..."));
+                }
+                if let Err(e) = authority_node_client
+                    .delete_member(ctx, identifier.clone())
+                    .await
                 {
-                    return Err(miette!(fmt_err!(
-                        "You need to use an enrolled identity to delete all the members from a project."
-                    )));
+                    warn!("Failed to delete member {}: {}", identifier, e);
+                } else {
+                    output.identifiers.push(identifier.clone());
                 }
-                authority_node_client
-                    .delete_all_members(ctx, identity.identifier())
-                    .await?;
-                opts.terminal.stdout().plain(fmt_ok!(
-                    "All identifiers except {} are no longer members of the Project.",
-                    color_primary(identity.identifier().to_string())
-                ));
             }
-            _ => unreachable!(),
+        } else {
+            unreachable!("Either a member or the --all flag should be set");
         }
 
+        opts.terminal
+            .stdout()
+            .plain(output.to_string())
+            .json_obj(&output)?
+            .write_line()?;
+
+        Ok(())
+    }
+}
+
+#[derive(Serialize)]
+struct DeleteMemberOutput {
+    project: String,
+    identifiers: Vec<Identifier>,
+}
+
+impl Display for DeleteMemberOutput {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self.identifiers.len() {
```

Comment:
Pattern matching on the slice would be a bit more elegant:
```
match self.identifiers[..] {
    [] => "There are no members",
    [identifier] => "Identifier {} ...",
    _ => "n members were deleted", 
}
```

Reply:
Oh I like that, I wasn't aware of this!